src/Store.js

```diff
@@ -388,12 +388,7 @@ class Store {
       return transaction(() =>
         json.data.map((dataObject) => {
           const { id, attributes = {}, relationships = {} } = dataObject
-          const ModelKlass = this.modelTypeIndex[type]
-          const record = new ModelKlass({
-            store,
-            relationships,
-            ...attributes
-          })
```

Comment:
1. This is erroneously not setting `id`
2. This duplicates code found in `createModel`